public/javascripts/showprojectscript.js

```diff
@@ -1,32 +1,69 @@
 //Function for the collapse all button
 function collapseAll() {
     var elements = document.getElementsByClassName("paddingCollapsible");
-    for (var x = 0; x < elements.length; x++) {
-        var content = elements[x].nextElementSibling;
-        if (content.style.display === "none") {
-            document.getElementById("compress").style.display = "block";
-            $.cookie(elements[x].id, "true");
-            document.getElementById("expand").style.display = "none";
-            content.style.display = "block";
-        } else {
-            content.style.display = "none";
-            $.cookie(elements[x].id, "false");
-            document.getElementById("expand").style.display = "block";
-            document.getElementById("compress").style.display = "none";
+    var buttonCollapseAll = document.getElementById('buttonCollapseAll');
+    var collapsed = buttonCollapseAll.value;
+    console.log(elements.length);
+    console.log(collapsed);
+    if (elements.length == collapsed) {
+        // all of them are collapsed, so uncollapse all
+        for (let element of elements) {
+            var id = element.id.substr(11,); // extract the id
+            uncollapse(id);
         }
-
+        buttonCollapseAll.value = 0;
     }
-    var deleteButtonCollapse = document.getElementsByClassName("button3");
-    for (var x = 0; x < deleteButtonCollapse.length; x++) {
-        var content = deleteButtonCollapse[x];
-        if (content.style.display === "none") {
-            content.style.display = "block";
-        } else {
-            content.style.display = "none";
+    else {
+        // at least one of them is collapsed, so collapse all
+        for (let element of elements) {
+            var id = element.id.substr(11,); // extract the id
+            collapse(id);
         }
+        buttonCollapseAll.value = elements.length;
     }
 }
 
+// helper functions for collapsing and uncollapsing neumes
+function collapse(id) {
+    var coll = document.getElementById("collapsible" + id);
+    var nameButton = document.getElementById('showName' + id);
+    var content = coll.nextElementSibling;
+    var deleteButtonCollapse = document.getElementById("deleteNeumeButton" + id);
+    var collapseAllButton = document.getElementById("buttonCollapseAll");
+
+    $.cookie(this.id, "false");
```

Comment:
Could you tell me how cookies are used in cress?

Reply:
They were used to store whether a neume is collapsed when the page refreshes (so that they stay collapsed) but not all the logic got ported over. Fixed in the new commit!